src/app/business/delfin/storages/storages.component.ts

```diff
@@ -0,0 +1,601 @@
+                element['capacity'].usage = percentUsage;
+                
+                element['volumes'] = [];
+                element['storagePools'] = [];
+                let vols = [];
+                let pools = [];
+                this.ds.getAllVolumes().subscribe((res)=>{
+                    this.allVolumes = res.json().volumes;
+                    this.allVolumes.forEach(volElement => {
+                        if(volElement['storage_id'] == element['id']){
+                            vols.push(volElement);
+                        }
+                    });
+                }, (error)=>{
+                    console.log("Something went wrong. Could not fetch Volumes.", element['id'], error)
+                });
+                
+                let capData = {
+                    labels: ['Used','Free'],
+                    datasets: [
+                        {
+                            data: [element['used_capacity'], element['free_capacity']],
+                            backgroundColor: [
+                                "#FF6384",
+                                "#45e800"
+                            ],
+                            hoverBackgroundColor: [
+                                "#FF6384",
+                                "#45e800"
+                            ]
+                        }]    
+                };
+
+                let opt = {
+                    legend:{
+                        position: 'right'
+                    }
+                }
+                element['capacityData'] = capData;
+                element['chartOptions'] = opt;
+                element['volumes'] = vols;
+                element['storagePools'] = pools;
+            });
+
+            // Invoke prepareTree() to prepare the tree structure for the widget
+            this.prepareTree(this.allStorages);
+            this.loading = false;
+
+        }, (error)=>{
+            console.log("Something went wrong. Could not fetch all storages", error);
+            this.loading = false;
+        })
+        
+    }
+
+    showChart(){
+        
+    }
+
+    getStorageArrayRawCapacity(storages){
+
+    }
+
+    getStorageArrayUsableCapacity(){
+
+    }
+
+    syncAllStorages(){
+
+    }
+
+    syncStorage(storageId){
+        this.ds.syncStorageById(storageId).subscribe((res)=>{
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: 'Sync request has been sent. Please check back in some time.'});
+            this.getAllStorages();
+        }, (error)=>{
+            this.msgs = [];
+            let errorMsg = 'Error Syncing device.' + error.error_msg;
+            this.msgs.push({severity: 'error', summary: 'Error', detail: error});
+            console.log("Something went wrong. Could not initiate the sync.", error);
+        });
+    }
+    getVolumesByStorage(storageId){
+        this.ds.getAllVolumes().subscribe((res)=>{
```

Comment:
Doubt: we have already API for getting single volume. Then why we are calling getAllVolumes(). We may call, ex: getAllVolumesById("storageId"). This will be very fast.

Reply:
Will fix this. When I started the work was not aware of the filter. Will fix this in the subsequent PRs.

Reply:
Raised an issue to track this. #396